File: 1674. Minimum Moves to Make Array Complementary/1674.myans.rs
impl Solution {
    pub fn min_moves(nums: Vec<i32>, limit: i32) -> i32 {
        let n = nums.len();
        let mut diff = vec![0; (2*limit+2) as usize];

        for i in 0..(n+1)/2 {
            let a = nums[i].min(nums[n-1-i]);
            let b = nums[i].max(nums[n-1-i]);

            diff[2] += 2;
            diff[(a+1) as usize] -= 1;
            diff[(a+b) as usize] -= 1;
            diff[(a+b+1) as usize] += 1;
            diff[(b+limit+1) as usize] += 1; 
        }

        let mut current = 0;
        let mut min_ops = i32::MAX;

        for i in 2..(2*limit+1) as usize {
            current += diff[i];
            min_ops = min_ops.min(current);
        }

        min_ops
    }
}

File: 1674. Minimum Moves to Make Array Complementary/1674.myans.go
func minMoves(nums []int, limit int) int {
    diff := make([]int, 2*limit+2)
    n := len(nums)

    for i := 0; i < (n+1)/2; i++ {
        a := min(nums[i], nums[n-1-i])
        b := max(nums[i], nums[n-1-i])

        diff[2] += 2
        diff[a+1] -= 1
        diff[a+b] -= 1
        diff[a+b+1] += 1
        diff[b+limit+1] += 1
    }

    min_ops := math.MaxInt
    current_ops := 0

    for i := 2; i < 2*limit+2; i++ {
        current_ops += diff[i]
        min_ops = min(min_ops, current_ops)
    }

    return min_ops
}